opm/simulators/wells/SingleWellState.hpp

```diff
@@ -20,20 +20,28 @@
 #ifndef OPM_SINGLE_WELL_STATE_HEADER_INCLUDED
 #define OPM_SINGLE_WELL_STATE_HEADER_INCLUDED
 
+#include <vector>
+
+#include <opm/simulators/wells/SegmentState.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/Well/Well.hpp>
 #include <opm/parser/eclipse/EclipseState/Schedule/Events.hpp>
 
 namespace Opm {
 
 class SingleWellState {
 public:
-    SingleWellState(bool is_producer, double temp);
+    SingleWellState(bool is_producer, std::size_t num_phases, double temp);
 
     Well::Status status{Well::Status::OPEN};
     bool producer;
     double bhp{0};
     double thp{0};
-    double temperature{};
+    double temperature;
```

Comment:
This makes `temperature` undefined in the `SingleWellState` unless a call site explicitly assigns a value.  Please revert.

Reply:
I'll revert sure enough, not intentional in any way. That said I do think the fringe feature temperature gets way to much attention/focus in the well code.

Reply:
Ehhh - second thought; the change to uninitialized was actually semi-intentional - to highlight the fact that the temperature value is just garbage. With `double temperature{}` it is well defined from a C++ standpoint, but from a physics/reservoir it still just garbage - although garbage.

Anyway - I'll put that :horse: back in the stable and revert!

Reply:
> the change to uninitialized was actually semi-intentional - to highlight the fact that the temperature value is just garbage.

I appreciate that, but by making the data member uninitialised you also create undefined behaviour in any operation that works on the structure as a whole.

> With `double temperature{}` it is well defined from a C++ standpoint

And that's needed.